python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -618,9 +593,18 @@ def request_resources(self, resources):
         if isinstance(resources, list):
             self.resource_demand_vector = resources
         else:
+            self.resource_demand_vector = []
             for resource, count in resources.items():
-                self.resource_requests[resource] = max(
-                    self.resource_requests[resource], count)
+                try:
```

Comment:
Can you comment/explain what this is doing/why it needed to be changed?

Reply:
self.resource_requests is not longer used, so this translated the previous single dict to resource_demand_vector.

Reply:
I removed the else branch, it's not used.

Reply:
I added it because some tests used the "Resources" and this dict.